src/transport/http2.rs

```diff
@@ -201,19 +167,104 @@ impl<In: RpcMessage, Out: RpcMessage> ServerChannel<In, Out> {
             .http2_max_frame_size(Some(MAX_FRAME_SIZE))
             .http2_max_send_buf_size(MAX_FRAME_SIZE as usize)
             .serve(service);
-        Ok((Self(accept_rx), server))
+
+        let (stop_tx, mut stop_rx) = mpsc::channel::<()>(1);
+        let server = server.with_graceful_shutdown(async move {
+            // If the sender is dropped this will also gracefully terminate the server.
+            stop_rx.recv().await;
+        });
+        tokio::spawn(server);
+
+        Ok(Self {
+            channel: accept_rx,
+            stop_tx,
+        })
     }
-}
 
-impl<In: RpcMessage, Out: RpcMessage> fmt::Debug for ServerChannel<In, Out> {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_tuple("ServerChannel").field(&self.0).finish()
+    /// Handles a single HTTP2 request.
+    ///
+    /// This creates the channels to communicate the (optionally streaming) request and
+    /// response and sends them to the [`ServerChannel`].
+    async fn handle_one_http2_request(
+        req: Request<Body>,
+        accept_tx: Sender<(Receiver<In>, Sender<Out>)>,
+    ) -> Result<Response<Body>, String> {
+        let (req_tx, req_rx) = flume::bounded::<In>(32);
+        let (res_tx, res_rx) = flume::bounded::<Out>(32);
+        accept_tx
+            .send_async((req_rx, res_tx))
+            .await
+            .map_err(|_e| "unable to send")?;
+
+        Self::spawn_recv_forwarder(req, req_tx);
+        let body = res_rx.into_stream().map(|out| {
+            let data = bincode::serialize(&out)?;
+            // TODO: check size
```

Comment:
It might be better if res_rx was a stream of Bytes and the serialization and checking was done before sending. Then the caller would get an immediate error when sending a too big msg, instead of just an error being logged here.

But that's for a later PR

Reply:
Additionally it would probably mean this code would be shared between the different transports I guess.  So that'd be another benefit.